terraform/node/k8s/deploy/main.tf

```diff
@@ -0,0 +1,172 @@
+
+      # Patch the staging image if this is running as part of release testing
+      if [ "${var.repository}" = "aws-otel-java-instrumentation" ]; then
+        RELEASE_TESTING_SECRET_NAME=release-testing-ecr-secret
+        kubectl delete secret -n sample-app-namespace --ignore-not-found $RELEASE_TESTING_SECRET_NAME
+        kubectl create secret -n sample-app-namespace docker-registry $RELEASE_TESTING_SECRET_NAME \
+          --docker-server=https://${var.release_testing_ecr_account}.dkr.ecr.us-east-1.amazonaws.com \
+          --docker-username=AWS \
+          --docker-password="$${TOKEN}"
+
+        yq eval '.spec.template.spec.imagePullSecrets += [{"name": "release-testing-ecr-secret"}]' -i frontend-service-depl.yaml
+        yq eval '.spec.template.spec.imagePullSecrets += [{"name": "release-testing-ecr-secret"}]' -i remote-service-depl.yaml
+      fi
+
+      echo "LOG: Applying sample app deployment files"
+      kubectl apply -f frontend-service-depl.yaml
+      kubectl apply -f remote-service-depl.yaml
+
+      # Wait for sample app to be reach ready state
+      sleep 10
+      kubectl wait --for=condition=Ready --request-timeout '5m' pod --all -n sample-app-namespace
+
+      # Emit main and remote service pod IP
+      echo "LOG: Outputting remote service pod IP to SSM using put-parameter API"
+      aws ssm put-parameter --region ${var.aws_region} --name main-service-ip-${var.test_id} --type String --overwrite --value $(kubectl get pods -n sample-app-namespace --selector=app=sample-app -o jsonpath='{.items[0].status.podIP}')
+      aws ssm put-parameter --region ${var.aws_region} --name remote-service-ip-${var.test_id} --type String --overwrite --value $(kubectl get pod --selector=app=remote-app -n sample-app-namespace -o jsonpath='{.items[0].status.podIP}')
+
+      # Wait a bit more in case the sample apps aren't ready yet
+      sleep 30
+
+      # Deploy the traffic generator
+      kubectl create deployment -n sample-app-namespace traffic-generator \
+        --image=$ACCOUNT.dkr.ecr.${var.aws_region}.amazonaws.com/e2e-test-resource:traffic-generator \
+        --replicas=1
+
+      # Patch it with ImagePull always policy so that it pulls the latest image from the ECR
+      kubectl patch deployment -n sample-app-namespace traffic-generator --patch '{"spec": {"template": {"spec": {"containers": [{"name": "e2e-test-resource", "imagePullPolicy": "Always"}]}}}}'
+      kubectl patch deployment traffic-generator -n sample-app-namespace --type='json' -p='[{"op": "add", "path": "/spec/template/spec/imagePullSecrets", "value": [{"name": "ecr-secret"}]}]'
+                
+      # Add the appropriate environment variables to the traffic generator
+      kubectl set env -n sample-app-namespace deployment/traffic-generator MAIN_ENDPOINT=$(kubectl get pods -n sample-app-namespace --selector=app=sample-app -o jsonpath='{.items[0].status.podIP}'):8000
```

Comment:
Seem like it's possible to set multiple env variables in one command. As `set env` on deployment will trigger the rolling update, it might be better to set env variables in one shot to trigger fewer pod restart?

Reply:
Good callout, made the improvement